src/realm/alloc_slab.cpp

```diff
@@ -867,6 +870,8 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
                 m_baseline = size;
                 m_initial_chunk_size = size;
                 m_file_mappings->m_first_additional_mapping = get_section_index(m_initial_chunk_size);
+
+                realm::util::encryption_read_barrier(m_file_mappings->m_initial_mapping, 0, sizeof(Header));
```

Comment:
As before - if this is not done then the header data will not be updated with the decrypted data.